Lib/unittest/test/testmock/testasync.py

```diff
@@ -130,37 +113,39 @@ def foo(): pass
         self.assertTrue(inspect.iscoroutinefunction(mock))
 
     def test_future_isfuture(self):
-        mock = AsyncMock(asyncio.Future())
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+        fut = asyncio.Future()
+        loop.stop()
+        loop.close()
+        mock = AsyncMock(fut)
         self.assertIsInstance(mock, asyncio.Future)
 
 
 class AsyncAutospecTest(unittest.TestCase):
     def test_is_AsyncMock_patch(self):
         @patch(async_foo_name, autospec=True)
         def test_async(mock_method):
-            self.assertTrue(isinstance(
-                             mock_method.async_method,
-                             AsyncMock))
-            self.assertTrue(isinstance(mock_method, MagicMock))
+            self.assertIsInstance(mock_method.async_method, AsyncMock)
+            self.assertIsInstance(mock_method, MagicMock)
 
         @patch(async_foo_name, autospec=True)
         def test_normal_method(mock_method):
-            self.assertTrue(isinstance(
-                                mock_method.normal_method,
-                                MagicMock))
+            self.assertIsInstance(mock_method.normal_method, MagicMock)
 
         test_async()
         test_normal_method()
 
+    def test_create_autospec_instance(self):
+        with self.assertRaises(RuntimeError):
+            create_autospec(async_func, instance=True)
```

Comment:
This test fails since the `RuntimeError` check is commented out.

Reply:
Ah yeah, sorry I was trying to remember the answer to your question on why it was needed. It might not be and I'm debating about taking it out, I think I don't fully understand the purpose of instance=True and what we would want the behavior to be if it is True with create_autospec on a async function.

Reply:
No problem, we can remove the restriction later if needed. I have added one more doctest suggestion that would make the CI green and this can be merged.